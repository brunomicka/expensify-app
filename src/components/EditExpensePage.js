import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onEdit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onRemove = () => {
        this.props.removeExpense({id : this.props.expense.id});
        this.props.history.push('/');
    }
    render() {
        return (<div>
            <h1>Edit Expense</h1>
            <ExpenseForm
                expense={this.props.expense}
                onSubmit={this.onEdit} />
            <button onClick={this.onRemove}>Remove</button>
        </div>);
    }
}

const mapDispatchToProps = (dispatch, props) =>
({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (id) => dispatch(removeExpense(id))
});


const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expese) => expese.id === props.match.params.id)
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);