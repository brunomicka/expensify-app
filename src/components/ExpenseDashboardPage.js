import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <br />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);


export default ExpenseDashboardPage;