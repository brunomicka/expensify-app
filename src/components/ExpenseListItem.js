import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({ description, amount, createdAt, note, id }) => {
    return (
        <div>
            <p><strong>Description:</strong> <Link to={`/edit/${id}`}>{description}</Link></p>
            <p>
                <strong>Amount:</strong> 
                {numeral(amount/100).format('$ 0,0.00')}
            </p>
            <p>
                <strong>Created At:</strong> 
                {moment(createdAt).format('MMMM Do, YYYY')}
            </p>
            <p><strong>Note:</strong> {note}</p>
            <hr />
        </div>
    )
};

export default ExpenseListItem;