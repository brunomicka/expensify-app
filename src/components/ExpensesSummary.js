import React from 'react';
import selectExpenses from '../selectors/expenses';
import getTotalExpenses from '../selectors/expenses-total';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expenseTotal}) => {

    return (
        <div>
            <h3>Viewing  {expensesCount}
                {expensesCount > 1 ? ' expenses ' : ' expense '}
             totalling {numeral(expenseTotal/100).format('$0,0.00')}</h3>
        </div>);

}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expenseTotal : getTotalExpenses(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
