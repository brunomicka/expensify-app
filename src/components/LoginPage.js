import React from 'react';
import { connect } from 'react-redux';
import { startLoginGoogle, startLoginTwitter } from '../actions/auth';

export const LoginPage = ({ startLoginGoogle, startLoginTwitter }) => {
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Expensify App</h1>
                <p>It's time to get your expenses under control.</p>
                <div className="box-layout__logins">
                    <button className="button" onClick={startLoginGoogle}>Login with Google</button>
                    <button className="button" onClick={startLoginTwitter}>Login with Twitter</button>
                </div>
            </div>
        </div>);
}

const mapDispatchToProps = (dispatch) => ({
    startLoginGoogle: () => dispatch(startLoginGoogle()),
    startLoginTwitter: () => dispatch(startLoginTwitter())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);