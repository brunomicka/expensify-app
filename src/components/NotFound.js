import {  Link } from 'react-router-dom';
import React from 'react';

const NotFound = () => (
    <div>404!
        <Link to="/">Go Home</Link>
    </div>);

export default NotFound;