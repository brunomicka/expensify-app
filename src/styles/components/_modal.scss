.ReactModalPortal>div {
    opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
    align-items: center;
    display: flex;
    justify-content: center;
    transition: opacity 200ms ease-in-out;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
    opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
    opacity: 0;
}

.modal {
    background: $dark-blue;
    color: white;
    max-width: 30rem;
    outline: none;
    padding: $l-size;
    text-align: center;
}

.modal__title {
    margin: 0 0 $m-size 0;
}

.modal__subtitle {
    font-weight: 700;
    font-size: $font-size-large;
}

.modal__body {
    font-size: 2rem;
    font-weight: 300;
    margin: 0 0 $s-size 0;
    word-break: break-all;
}

.modal__buttons {
    display: flex;
    justify-content: space-around;

    button{
        padding: $m-size;
    }
}