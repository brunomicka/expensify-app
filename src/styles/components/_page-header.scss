.page-header {
    background: $off-white;
    margin-bottom: $l-size;
    padding: $l-size 0;
}

.page-header__actions {
    margin-top: $m-size;

}

.page-header__title {
    margin: 0;
    font-weight: 300;

    span {
        font-weight: 700;
    }
}

.page-header__subtitle {
    font-style: italic;

    span {
        font-weight: 700;
    }
}