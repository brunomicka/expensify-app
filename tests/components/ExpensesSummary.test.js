import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import getTotalExpenses from '../../src/selectors/expenses-total';
import { ExpensesSummary } from '../../src/components/ExpensesSummary';

test('Should render ExpensesSummary with a singular expense correctly', () => {
    const singularExpense = [expenses[1]];
    const wrapper = shallow(<ExpensesSummary
        expensesCount={singularExpense.length}
        expenseTotal={getTotalExpenses(singularExpense)} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary with multiple expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary
        expensesCount={expenses.length}
        expenseTotal={getTotalExpenses(expenses)} />);
    expect(wrapper).toMatchSnapshot();
});