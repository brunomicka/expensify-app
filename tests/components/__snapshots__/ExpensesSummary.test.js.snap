// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with a singular expense correctly 1`] = `
<div>
  <h3>
    Viewing  
    1
     expense 
    totalling 
    $1,095.00
  </h3>
</div>
`;

exports[`Should render ExpensesSummary with multiple expenses correctly 1`] = `
<div>
  <h3>
    Viewing  
    3
     expenses 
    totalling 
    $1,141.95
  </h3>
</div>
`;
